### PR TITLE
Implement request search by target and source for AJAX datatables

### PR DESCRIPTION
**Pull request by ChrisBr**

### CODE REVIEW

File: src/api/app/assets/javascripts/webui/application/requests_table.js [hunk 1]
```diff
@@ -10,9 +10,10 @@ $( document ).ready(function() {
       order: [[0,'desc']],
       info: false,
       columnDefs: [
-        // We only allow ordering for created, requester and priority.
-        // Columns: created, source, target, requester, type, priority.
-        { orderable: false, targets: [1,2,4,6,] }
+        // We dont allow ordering for type and request link.
+        // Columns: created, source, target, requester, type, priority, request link.
+        // First column has index 0.
+        { orderable: false, targets: [4,6,] }
```
Comment (evanrolfe) [inline src/api/app/assets/javascripts/webui/application/requests_table.js]: [Nitpick] Trailing comma in the array?

File: src/api/app/models/bs_request/data_table/params_parser.rb [hunk 2]
```diff
@@ -43,16 +42,22 @@ def order_params
       def sort_column
```
Comment (evanrolfe) [inline src/api/app/models/bs_request/data_table/params_parser.rb]: [Nitpick] change this to `sort_columns` now that it returns an array.

**Comment by evanrolfe** [comment]
Harikiri is complaining about potential SQL injection in `app/models/bs_request/data_table/find_for_user_query.rb` but thats not a danger since params[:sort] is generated from pre defined values.

**Comment by ChrisBr** [comment]
@evanrolfe hmm indeed, let me check if I can rewrite it that Hakiri does not complain ...

**Comment by ChrisBr** [comment]
@evanrolfe fixed your suggestions! I think the hakiri issue was caused as we didn't use ``@`` for the instance variable so Hakiri thought we use the ActionController params (that was also the reason Hakiri only reported the user datatable but not package / project data tables)

**Comment by ChrisBr** [comment]
[Review list no longer sortable by 'source' #3582](https://trello.com/c/Mo9tKD7G/487-review-list-no-longer-sortable-by-source-3582)

**Pull request merged**